### DISCUSSION TITLE
Share business logic / extensions with front

### DISCUSSION THREAD

**Discussion by Disregard-Therest** [serverpod/serverpod#1716] [Q&A] [answered]
Hi, guys, what is the correct way to share some functionality and helpers across back and front?

I mean, for example, enum extensions.

**Comment by SandPod** [comment, marked as answer]
Hey @Disregard-Therest!

### Sharing business logic
To share business logic between the server and the client the correct way is to create another dart project, that could be called `<you_project>_shared` and then import it in both placed.

Serverpod internally shares code in this way through the `serverpod_shared` package.

### Sharing model operations
Unfortunately, the workaround for sharing model operations is a bit complex. You can find discussions and instructions on it here: https://github.com/serverpod/serverpod/issues/1256.

The reason it is complicated is that the server and the client have different implementations for the models.

We are planning on fixing this in the future with the suggested solution in the issue, automatically generating a `shared` project that include an implementation of the shared model between the server and the client so that extensions and model operations can easily be shared.

**Reply by Disregard-Therest** [reply]
Great, thank you!
